crates/vizia_core/src/context/mod.rs

```diff
@@ -355,19 +375,34 @@ impl Context {
     }
 
     /// Add a font from memory to the application.
-    pub fn add_font_mem(&mut self, name: &str, data: &[u8]) {
-        // TODO - return error
-        if self.resource_manager.fonts.contains_key(name) {
-            println!("Font already exists");
-            return;
-        }
-
-        self.resource_manager.fonts.insert(name.to_owned(), FontOrId::Font(data.to_vec()));
+    pub fn add_fonts_mem(&mut self, data: &[&[u8]]) {
+        self.cosmic_context.take_buffers();
+        replace_with_or_abort(&mut self.cosmic_context, |mut ccx| {
+            let buffers = ccx.take_buffers();
+            let (locale, mut db) = ccx.into_font_system().into_locale_and_db();
+            for font_data in data {
+                db.load_font_data(Vec::from(*font_data));
+            }
+            let mut new_ccx = CosmicContext::new_from_locale_and_db(locale, db);
+            for (entity, lines) in buffers {
+                new_ccx.with_buffer(entity, move |buf| {
+                    buf.lines = lines
+                        .into_iter()
+                        .map(|line| BufferLine::new(line, AttrsList::new(Attrs::new())))
+                        .collect();
+                });
+            }
+            new_ccx
+        });
     }
```

Comment:
I guess this is required because cosmic doesn't provide a way to add fonts after the context has been created?

Reply:
yep - this is a workaround for https://github.com/pop-os/cosmic-text/issues/57